### DISCUSSION TITLE
Why does PackerInstall keep deleting things?

### DISCUSSION THREAD

**Discussion by GideonWolfe** [wbthomason/packer.nvim#540] [Q&A]
So I'm trying to put together my `lua` `neovim` configuration, and I'm adding some plugins to help my workflow. 

The issue is, I'll be using some plugin for a while without issue, and then when I try to add a completely unrelated plugin and run `:PackerInstall`, I'm prompted like this

![image](https://user-images.githubusercontent.com/32942052/128644933-43c16723-8d0e-4c23-b980-c37f49632e75.png)

The only thing I added to my Packer config was `galaxyline.nvim`, and it's not even close to that location in the file. I certainly don't want to remove `friendly-snippets`.

It won't proceed without agreement, and the last time I accepted it, my perfectly working `compe` + `luasnips` configuration starts throwing errors. It's kind of like 2 steps forward, one step back.

**Comment by wbthomason** [comment]
`packer` should only request to delete a plugin if it's not in your currently loaded plugin specification. It should also allow you to proceed in the install even if you reply "N" to the cleaning prompt. Usually when an unexpected deletion prompt happens, it's because of Lua module caching - the cached version of your plugin specification module doesn't include the plugin `packer` is asking to delete.

Can you reproduce the issue consistently, across a restart of Neovim? As well, when you say "it won't proceed without agreement" - what specifically happens when you reply to the prompt with "N"?

**Reply by GideonWolfe** [reply]
On this particular occasion I said yes and it then just brought up a window saying it installed successfully... So I guess that's good? Just a bit confusing haha

**Reply by wbthomason** [reply]
So, whether you said yes or no to the cleaning, the install should still have proceeded. The question is more why `packer` tried to clean a plugin you wanted installed - like I suggested in my earlier response, this is usually due to running an operation with a cached plugin specification module.
